### ISSUE TITLE
Control log level

### ISSUE THREAD

**Issue by spartanhooah** [tomtom-international/goji-http-client#140]
My team has used this library to write integration tests since `http-builder-ng` got put on ice. One thing that we are missing currently compared to that library is detailed HTTP traffic logs. E.g. seeing request headers and content in the test (Gradle in our case) logs. Is this something that is already possible and I just need to learn how, or is it something that would need to be added to this project?

**Comment by artamonovkirill** [comment]
Hi, @spartanhooah 

Thanks for raising this!

No, the library itself doesn't produce any logs.
There are [some logging options](https://hc.apache.org/httpcomponents-client-4.5.x/logging.html) in the underlying `org.apache.httpcomponents:httpclient`, but I guess the best would be to either abstract it away from the library user or introduce own logging in `goji-http-client`.

Do you have any specific requirements in mind qua logs format?

**Comment by spartanhooah** [comment]
I think at minimum the ability to display incoming and outgoing headers, and separately incoming and outgoing request/response bodies would go a long way towards making debugging friendlier.

**Comment by artamonovkirill** [comment]
@spartanhooah, please let me know what you think of:
```
2023-09-23 14:54:20 INFO POST http://localhost:53611/freezer
2023-09-23 14:54:20 INFO     body: ice-cream
2023-09-23 14:54:20 INFO => response: 200
2023-09-23 14:54:20 INFO     headers: [Matched-Stub-Id: f87222a3-758e-4604-a4df-da9890d8b3f7, Transfer-Encoding: chunked]
...
2023-09-23 14:54:20 INFO POST http://localhost:53611/freezer
2023-09-23 14:54:20 INFO     body: <src/test/resources/ice-cream.txt file content omitted>
2023-09-23 14:54:20 INFO => response: 200
2023-09-23 14:54:20 INFO     headers: [Matched-Stub-Id: 446caf18-069e-46ec-9977-b57d5ae75108, Transfer-Encoding: chunked]
...
2023-09-23 14:54:20 INFO POST http://localhost:53611/freezer
2023-09-23 14:54:20 INFO     body: {"type":"ice-cream"}
2023-09-23 14:54:20 INFO => response: 200
2023-09-23 14:54:20 INFO     headers: [Matched-Stub-Id: b3e9f757-d959-47ec-94ee-f8425a99ec48, Transfer-Encoding: chunked]
...
2023-09-23 14:54:20 INFO GET http://localhost:53611/freezer
2023-09-23 14:54:20 INFO     headers: [shelve: top]
2023-09-23 14:54:20 INFO => response: 200
2023-09-23 14:54:20 INFO     headers: [Matched-Stub-Id: a921564a-5bd7-44fd-afd1-f482cc6541f0, Transfer-Encoding: chunked]
...
2023-09-23 14:54:20 INFO GET http://localhost:53611/freezer
2023-09-23 14:54:20 INFO => response: 200
2023-09-23 14:54:20 INFO     headers: [Matched-Stub-Id: 0b47df67-a423-4aef-9ef2-dea4319f91f4, Transfer-Encoding: chunked]
2023-09-23 14:54:20 INFO     body: {"contents": ["ice-cream"]}
...
2023-09-23 14:54:21 INFO GET http://localhost:53611/freezer?foo=bar&foo=baz
2023-09-23 14:54:21 INFO => response: 200
2023-09-23 14:54:21 INFO     headers: [Matched-Stub-Id: d4b1b9ed-97ec-4247-8b02-a5c41629ea63, Transfer-Encoding: chunked]
```
? Thanks in advance.

**Comment by spartanhooah** [comment]
That looks great!

More to the original issue, I have noticed that in some projects, I get debug logs from `org.apache.http.*`, but in others I don't. I have compared the projects to try to find any relevant differences, but I can't. They both use the same version of this library, they're both using a Gradle wrapper around Gradle 7.6. One thing I have noticed is that neither project logs the debug logs when I run the tests from the terminal. However, one project does if I run the tests using IntelliJ. It doesn't matter if I tell IntelliJ to run tests using IntelliJ or Gradle; that project always logs the debug logs. The other project never does. It's infuriating, because I'm evidently creating more work for you when really it must be some configuration difference on my end that I just can't pin down.
